library/Icingadb/Model/DependencyNode.php

```diff
@@ -43,10 +48,59 @@ public function getColumns(): array
             'id',
             'host_id',
             'service_id',
-            'redundancy_group_id'
+            'redundancy_group_id',
+            'name' => new Expression(
+                'CASE WHEN %s IS NOT NULL THEN %s WHEN %s IS NOT NULL THEN %s ELSE %s END',
+                ['service_id', 'service.display_name', 'host_id', 'host.display_name', 'redundancy_group.display_name']
+            ),
+            'severity' => new Expression(
+                'CASE WHEN %s IS NOT NULL THEN %s WHEN %s IS NOT NULL THEN %s ELSE %s END',
+                [
+                    'service_id',
+                    'service.state.severity',
+                    'host_id',
+                    'host.state.severity',
+                    'redundancy_group.state.failed'
+                ]
+            ),
+            'state' => new Expression(
+                'CASE WHEN %s IS NOT NULL THEN %s WHEN %s IS NOT NULL THEN %s ELSE %s END',
+                [
+                    'service_id',
+                    'service.state.soft_state',
+                    'host_id',
+                    'host.state.soft_state',
+                    'redundancy_group.state.failed'
+                ]
+            ),
+            'last_state_change' => new Expression(
+                'CASE WHEN %s IS NOT NULL THEN %s WHEN %s IS NOT NULL THEN %s ELSE %s END',
+                [
+                    'service_id',
+                    'service.state.last_state_change',
+                    'host_id',
+                    'host.state.last_state_change',
+                    'redundancy_group.state.last_state_change'
+                ]
+            )
```

Comment:
I think you added this to be able to provide sort rules in the members and children views. Please don't do this with expressions. Add a sort rule with multiple columns instead for each. `severity => ['columns' => […], 'dir' => 'desc']`